### PR TITLE
Update isReportMessageAttachment logic

### PR DESCRIPTION
**Pull request by sobitneupane**
### Details

Solves the issue "When [Attachment] is sent as the message, 'Edit Comment' option does not appear in contextMenu "

### Fixed Issues

$ https://github.com/Expensify/App/issues/7739

### Tests

1. Send attachment and hover over attachment to see only three options ('Copy to clipboard', 'Mark as unread', 'Delete comment')
2. Send message text '[Attachment]' and hover over the message to see all four options ('Copy to clipboard', 'Mark as unread', 'Edit comment', 'Delete comment')
- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I verified the PR has a small number of commits behind `main`
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [ ] I clearly indicated the environment tests should be run in (Staging vs Production)
- [x] I wrote testing steps that cover success & fail scenarios (if applicable)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests & veryfy they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors related to changes in this PR
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I added comments when the code was not self explanatory
    - [ ] I put all copy / text shown in the product in all `src/languages/*` files (if applicable)
    - [ ] I followed proper naming convention for platform-specific files (if applicable)
    - [ ] I followed style guidelines (in [`Styling.md`](https://github.com/Expensify/App/blob/main/STYLING.md)) for all style edits I made
    - [x] I followed the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs))
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] I corroborated the UI performance was not affected (the performance is the same than `main` branch)
- [ ] If I created a new component I verified that a similar component doesn't exist in the codebase

#### PR Reviewer Checklist
- [ ] I verified the PR has a small number of commits behind `main`
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the testing environment is mentioned in the test steps
- [ ] I verified testing steps cover success & fail scenarios (if applicable)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors related to changes in this PR
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified comments were added when the code was not self explanatory
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files (if applicable)
    - [ ] I verified proper naming convention for platform-specific files was followed (if applicable)
    - [ ] I verified [style guidelines](https://github.com/Expensify/App/blob/main/STYLING.md) were followed
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components are not impacted by changes in this PR (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] I verified the UI performance was not affected (the performance is the same than `main` branch)
- [ ] If a new component is created I verified that a similar component doesn't exist in the codebase

### QA Steps

1. Send attachment and hover over attachment to see only three options ('Copy to clipboard', 'Mark as unread', 'Delete comment')
2. Send message text '[Attachment]' and hover over the message to see all four options ('Copy to clipboard', 'Mark as unread', 'Edit comment', 'Delete comment')
- [x] Verify that no errors appear in the JS console

#### Web

https://user-images.githubusercontent.com/25876548/158072519-4d49505b-fc26-4859-b59d-1f4754236eb5.mov

#### Mobile Web

https://user-images.githubusercontent.com/25876548/158072610-c4523999-a3ad-4520-86d5-ca7703503298.mov

#### Desktop

https://user-images.githubusercontent.com/25876548/158072560-ef187b34-dfb7-408e-8de9-02e71b48fcde.mov

#### iOS

https://user-images.githubusercontent.com/25876548/158072574-a87f7730-d86e-4271-b788-4e32726eec55.mov

#### Android

https://user-images.githubusercontent.com/25876548/158072615-2fd38744-0ee0-4973-9c09-41058d858db6.mov

### CODE REVIEW

**Review by parasharrajat — CHANGES_REQUESTED** [review]
Do you see a pattern here that can be optimized?

File: src/libs/reportUtils.js [hunk 1]
```diff
@@ -26,10 +26,11 @@ function getReportParticipantsTitle(logins) {
  * Check whether a report action is Attachment is not.
  *
  * @param {Object} reportMessageText report action's message as text
+ * @param {Object} reportMessageHtml report action's message as html
  * @returns {Boolean}
  */
-function isReportMessageAttachment(reportMessageText) {
-    return reportMessageText === '[Attachment]';
+function isReportMessageAttachment(reportMessageText, reportMessageHtml) {
+    return reportMessageText === '[Attachment]' && reportMessageHtml !== '[Attachment]';
```
Comment (Luke9389) [inline src/libs/reportUtils.js:33]: could put the `lodashGet` in here rather than passing it in as an argument.

Comment (Luke9389) [reply src/libs/reportUtils.js:33]: Whoops! Sorry @parasharrajat, didn't see [this](https://github.com/Expensify/App/pull/8117#pullrequestreview-908169352) before leaving this comment 😅

Comment (parasharrajat) [reply src/libs/reportUtils.js:33]: No worries.

**Comment by sobitneupane** [comment]
> Do you see a pattern here that can be optimized?

`reportMessageText` and `reportMessageHtml` are obtained from different variables in different cases.
In reportUtils.js, ContextMenuAction.js and Report.js, they are obtained from `reportAction` but
in OptionsListUtils.js, they are obtained from `report`. 

So, they are extracted from variable(`reportAction` or `report`) before passing to the function `isReportMessageAttachment()`.

**Comment by parasharrajat** [comment]
Ok. Awesome. let's do it. pass a single param to that function that checks both values.

**Comment by sobitneupane** [comment]
> Ok. Awesome. let's do it. pass a single param to that function that checks both values.

Is this what you are suggesting?
```
function isReportMessageAttachment(reportMessage) {
    return lodashGet(reportMessage, 'text', '') === '[Attachment]' && lodashGet(reportMessage, 'html', '') !== '[Attachment]';
}
```
```
isReportMessageAttachment({text: lodashGet(reportAction, ['message', 0, 'text'], ''), html: lodashGet(reportAction, ['message', 0, 'html'], '')});
```

File: src/libs/reportUtils.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ function canEditReportAction(reportAction) {
     return reportAction.actorEmail === sessionEmail
         && reportAction.reportActionID
         && reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.ADDCOMMENT
-        && !isReportMessageAttachment(lodashGet(reportAction, ['message', 0, 'text'], ''));
+        && !isReportMessageAttachment({text: lodashGet(reportAction, ['message', 0, 'text'], ''), html: lodashGet(reportAction, ['message', 0, 'html'], '')});
```
Comment (parasharrajat) [inline src/libs/reportUtils.js:63]:
```suggestion
        && !isReportMessageAttachment(lodashGet(reportAction, ['message', 0], {}));
```

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/pages/home/report/ContextMenu/ContextMenuActions.js [hunk 3]
```diff
@@ -58,7 +58,7 @@ export default [
 
             const isAttachment = _.has(reportAction, 'isAttachment')
                 ? reportAction.isAttachment
-                : ReportUtils.isReportMessageAttachment(text);
+                : ReportUtils.isReportMessageAttachment(message);
```
Comment (parasharrajat) [inline src/pages/home/report/ContextMenu/ContextMenuActions.js:61]: Are you sure that message will always be a Object here?

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/pages/home/report/ContextMenu/ContextMenuActions.js (hunk 3 above)
Comment (sobitneupane) [reply src/pages/home/report/ContextMenu/ContextMenuActions.js:61]: I have tested in different scenarios and it does return object. But One improvement that can be made is:
```diff
-            const message = _.last(lodashGet(reportAction, 'message', null));
+            const message = _.last(lodashGet(reportAction, 'message', [{}]));

```

Comment (parasharrajat) [reply src/pages/home/report/ContextMenu/ContextMenuActions.js:61]: Yup. Good catch. lets do this.

File: src/libs/reportUtils.js [hunk 4]
```diff
@@ -25,11 +25,11 @@ function getReportParticipantsTitle(logins) {
 /**
  * Check whether a report action is Attachment is not.
  *
- * @param {Object} reportMessageText report action's message as text
+ * @param {Object} reportActionMessage report action's message as text and html
```
Comment (parasharrajat) [inline src/libs/reportUtils.js:28]: Also, I feel like we need a more sophisticated comment now. we should tell why are we ignoring the message when HTML is `[Attachment]`.

Comment (sobitneupane) [reply src/libs/reportUtils.js:28]: Does this sound good?
```diff
/**
 * Check whether a report action is Attachment or not.
+ * If both text and html of reportActionMessage is [Attachment], text '[Attachment]' not an attachment is the message.
 *
 * @param {Object} reportActionMessage report action's message as text and html
 * @returns {Boolean}
 */
```

Comment (parasharrajat) [reply src/libs/reportUtils.js:28]: Ignore messages containing `[Attachment]` as the main content. Attachments are actions with only text as `[Attachment]`.

**Review by parasharrajat — DISMISSED** [review]
LGTM. 

cc: @Luke9389 

#### PR Reviewer Checklist
- [x] I verified the PR has a small number of commits behind `main`
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the testing environment is mentioned in the test steps
- [x] I verified testing steps cover success & fail scenarios (if applicable)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors related to changes in this PR
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified comments were added when the code was not self explanatory
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files (if applicable)
    - [x] I verified proper naming convention for platform-specific files was followed (if applicable)
    - [x] I verified [style guidelines](https://github.com/Expensify/App/blob/main/STYLING.md) were followed
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components are not impacted by changes in this PR (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified the UI performance was not affected (the performance is the same than `main` branch)
- [x] If a new component is created I verified that a similar component doesn't exist in the codebase

:ribbon: :eyes: :ribbon:  C+ reviewed

**Review by Luke9389 — COMMENTED** [review]
Lookin' good.
Just one NAB. Up to you if you'd like to implement that.

**Comment by sobitneupane** [comment]
> Lookin' good.
> Just one NAB. Up to you if you'd like to implement that.

@Luke9389 Made the suggested change.

**Review by Luke9389 — APPROVED** [review]

**Pull request merged**